### PR TITLE
Version checker

### PR DESCRIPTION
**Pull request by ejwessel**
## Description

Quick script to aid to keeping the versions in sync before publishing.
Added to circle ci to prevents merge to master when versions are out of sync.

Note: this does not update the package versions, but 

Quick description:

- [X] N/A
- [ ] New features
- [ ] Bug fixes
- [ ] Typo/small tweaks

## Changes

Implementation details. These are mainly to help your fellow developers.

## Tests
```go
source venv/bin/activate; python scripts/identify_dependencies.py@airswap/types 0.4.3

 - @airswap/tokens -> 0.1.3
 - @airswap/order-utils -> 0.3.8

@airswap/wrapper 0.4.1
 - @airswap/swap -> 2.3.2
 - @airswap/tokens -> 0.1.3
 - @airswap/order-utils -> 0.3.8
 - @airswap/test-utils -> 0.1.1

@airswap/swap 2.3.2
 - @airswap/types 0.4.2 -> 0.4.3  Version Mismatch
 - @airswap/types -> 0.4.3
 - @airswap/tokens -> 0.1.3
 - @airswap/order-utils -> 0.3.8
 - @airswap/test-utils -> 0.1.1

@airswap/delegate 0.5.1
 - @airswap/indexer -> 0.4.2
 - @airswap/swap -> 2.3.2
 - @airswap/order-utils -> 0.3.8
 - @airswap/test-utils -> 0.1.1

@airswap/delegate-factory 0.3.2
 - @airswap/delegate -> 0.5.1
 - @airswap/order-utils -> 0.3.8
 - @airswap/test-utils -> 0.1.1

@airswap/index 0.3.1
 - @airswap/order-utils -> 0.3.8
 - @airswap/test-utils -> 0.1.1
 - @airswap/tokens -> 0.1.3

@airswap/tokens 0.1.3

@airswap/indexer 0.4.2
 - @airswap/index -> 0.3.1
 - @airswap/order-utils -> 0.3.8
 - @airswap/test-utils -> 0.1.1
 - @airswap/types 0.4.2 -> 0.4.3  Version Mismatch
 - @airswap/types -> 0.4.3

@airswap/test-utils 0.1.1

@airswap/order-utils 0.3.8
 - @airswap/swap -> 2.3.2
 - @airswap/tokens -> 0.1.3
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "lint": "yarn eslint \"./**/*.js\"",
     "ganache": "ganache-cli -p 8545 --gasLimit 0xfffffffffff --time '2017-05-10T00:00:00+00:00' -s 0",
     "publish": "lerna publish",
-    "test": "yarn clean && yarn compile && lerna run test --concurrency=1"
+    "test": "yarn clean && yarn compile && lerna run test --concurrency=1",
+    "check_dep_graph" : "source venv/bin/activate; python scripts/identify_dependencies.py"
```
Comment (alicevhenshaw) [inline package.json]: venv is in .gitignore so this command won't work unless people happen to have their venv setup the same way you do locally

Comment (poiresel) [reply package.json]: rather just remove source venv/bin/activate and make a note of usage in README.md or in the python_scripts directory.

Comment (poiresel) [reply package.json]: circleci may need to install python and virtualenv if this needs to be done.

**Comment by ejwessel** [comment]
going to move it to circle

File: package.json (hunk 1 above)
Comment (ejwessel) [reply package.json]: correct. working with link on that now.

File: .circleci/config.yml [hunk 2]
```diff
@@ -24,6 +24,16 @@ references:
...
+      - run: 
+          name: Check Dependencies
+          command: |
+            sudo apt install python3-pip
+            pip3 install --user virtualenv
+            virtualenv venv
+            python3 --version
+            source venv/bin/activate
+            pip3 install pathlib
+            python3 scripts/identify_dependencies.py
```
Comment (poiresel) [inline .circleci/config.yml]: do you have to do this? instead just install in the global context ie

sudo apt install python3-pip
pip3 install pathlib
 python3 scripts/identify_dependencies.py

File: scripts/identify_dependencies.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                    # skip if we don't see the packages we care about
+                    if not self.contains_packages(declared_name):
+                        continue
+
+                    # check version against declared version
+                    expected_version = self.dependency_graph[declared_name]['version']
+                    declared_version = declared_ver
+                    if declared_version != expected_version:
+                        stable = False
+                        print(" - %s %s -> %s %s Version Mismatch %s" % (declared_name, declared_version, expected_version, bcolors.FAIL, bcolors.ENDC))
```
Comment (poiresel) [inline scripts/identify_dependencies.py]: recommend to use f strings 
f" - {declared_name} {declared_version} -> {expected_version} {bcolors.FAIL} Version Mismatch {bcolors.ENDC}" especially if you're using  python3

File: package.json (hunk 1 above)
Comment (alicevhenshaw) [reply package.json]: Please can you include a `requirements.txt` of the required venv setup to run the script? Such that after activating a venv locally you can just run `pip3 install -r requirements.txt` to get all requirements needed, and then run the script?

Comment (alicevhenshaw) [reply package.json]: I think the command is just `pip freeze > requirements.txt`

Comment (alicevhenshaw) [reply package.json]: extra space in `"check_dep_graph" : ` before the colon

Comment (alicevhenshaw) [reply package.json]: do you think that this command should now include `pip3 install -r requirements.txt` between the venv activate and running the script?

**Pull request merged**